spring-cloud-alibaba-starters/spring-cloud-starter-alibaba-appactive/src/main/java/com/alibaba/cloud/appactive/provider/URIRegister.java

```diff
@@ -46,68 +47,82 @@ public final class URIRegister {
-				}
+	public static void collectUris(List<FilterRegistrationBean<? extends Filter>> beanList) {
+		if (CollectionUtils.isEmpty(beanList)) {
+			return;
+		}
+		List<ServiceMeta> serviceMetaList = new LinkedList<>();
+		boolean hasWildChar = false;
+		for (FilterRegistrationBean<? extends Filter> filterRegistrationBean : beanList) {
+			Filter filter = filterRegistrationBean.getFilter();
+			if (filter == null) {
+				continue;
 			}
-			if (CollectionUtils.isNotEmpty(serviceMetaList)) {
-				if (!hasWildChar) {
-					// 保证所有 service(app+uri) 都纳入管理，不然不好做缓存管理
-					ServiceMeta serviceMeta = new ServiceMeta(MATCH_ALL,
-							ResourceActiveType.NORMAL_RESOURCE_TYPE);
-					serviceMetaList.add(serviceMeta);
-				}
-				serviceMetaObject = new ServiceMetaObject();
-				Collections.sort(serviceMetaList);
-				serviceMetaObject.setServiceMetaList(serviceMetaList);
-				String meta = JSON.toJSONString(serviceMetaList);
-				serviceMetaObject.setMeta(meta);
-				String md5 = DigestUtils.md5Hex(meta.getBytes(StandardCharsets.UTF_8));
-				serviceMetaObject.setMd5OfList(md5);
+			Collection<String> urlPatterns = filterRegistrationBean.getUrlPatterns();
+			if (filter instanceof CoreServiceFilter) {
+				hasWildChar = expandAndDetermine(serviceMetaList, hasWildChar, urlPatterns,
+						ResourceActiveType.UNIT_RESOURCE_TYPE);
+			} else if (filter instanceof GlobalServiceFilter) {
+				hasWildChar = expandAndDetermine(serviceMetaList, hasWildChar, urlPatterns,
+						ResourceActiveType.CENTER_RESOURCE_TYPE);
+			} else if (filter instanceof GeneralServiceFilter) {
+				hasWildChar = expandAndDetermine(serviceMetaList, hasWildChar, urlPatterns,
+						ResourceActiveType.NORMAL_RESOURCE_TYPE);
 			}
 		}
+		if (CollectionUtils.isEmpty(serviceMetaList)) {
+			return;
+		}
+		if (!hasWildChar) {
+			// 保证所有 service(app+uri) 都纳入管理，不然不好做缓存管理
+			expand(serviceMetaList, MATCH_ALL, ResourceActiveType.NORMAL_RESOURCE_TYPE);
+		}
+		initServiceMetaObject(serviceMetaList);
+	}
+
+	/**
+	 * initialize {@link #serviceMetaObject} based on {@link ServiceMeta} list.
+	 * @param serviceMetaList list needed for initialization
+	 */
+	private static void initServiceMetaObject(List<ServiceMeta> serviceMetaList) {
+		serviceMetaObject = new ServiceMetaObject();
+		Collections.sort(serviceMetaList);
+		serviceMetaObject.setServiceMetaList(serviceMetaList);
+		String meta = JSON.toJSONString(serviceMetaList);
+		serviceMetaObject.setMeta(meta);
+		String md5 = DigestUtils.md5Hex(meta.getBytes(StandardCharsets.UTF_8));
+		serviceMetaObject.setMd5OfList(md5);
+	}
+
+	/**
+	 * Expand {@link ServiceMeta} list while determining whether it is a wild char.
+	 * @param serviceMetaList extended list
+	 * @param hasWildChar keyword to be determined
+	 * @param urlPatterns looped list
+	 * @param resourceActiveType attribute of {@link ServiceMeta}
+	 * @return is that a wild char
+	 */
+	private static boolean expandAndDetermine(List<ServiceMeta> serviceMetaList, boolean hasWildChar,
```

Comment:
I think called `collectServiceMeta` is better. this is the purpose of the logic.